Phpstan level 8

fruitl00p opened the pull request:
This adds various updates to reach phpstan level 8

fruitl00p committed:
feat(phpcs): make everyting final

fruitl00p committed:
feat(phpcs): protecteds are now private

fruitl00p committed:
feat(phpcs): return types ftw

fruitl00p committed:
feat(phpcs): phpdoc cleanup

fruitl00p committed:
feat(phpcs): phpstan lvl8

By defining the various shapes of the return types / params in docblocks

mishavantol commented:
This one doesn't blend. Sometimes you need a custom entity which extends one of these base classes. E.g. the NVR uses a custom Entity/Address which extends this Address to add companyTitle. Removing the ability to use these classes this way limits the usability of the package imo.

fruitl00p commented:
good point. In that case, we should add a config file to make sure the `final`-rule isn't adhered / enforced. As this is a library with that specific intent...

fruitl00p committed:
fix(style): fixes whitespace in phpstan docs

fruitl00p committed:
fix(style): the private constructor is now final

The class isn't final, but the constructor is so the static::create()-method wont fatal
(according to https://github.com/phpstan/phpstan/issues/2963)

fruitl00p commented:
@mishavantol the newer commits should fix the gripe ;)

fruitl00p committed:
fix(style): databag properties themselves should be protected to allow extension

fruitl00p committed:
feat(phpcs): phpstan lvl8 for the tests too!
